vtex-sites/gatsby.store
# feat: integrate with VTEX headless CMS

## What's the purpose of this pull request?

This PR adds CMS capabilities to `gatsby.store`. It addresses mostly parts of the [CMS Spike](https://vtex-dev.atlassian.net/browse/FSSS-352). This PR:

- Integrates `gatsby.store` with `storeframework` vtex store;
- Renders one component using CMS (global alert)
- Renders one page using cms (/cms)
- Adds page rendered to lighthouse tests

The script to remove all cms related code is not done but will be tackled later.

## How does it work?

Firstly, I followed the [cms tutorial](https://www.faststore.dev/tutorials/cms-overview) to integrate `gatsby.store` with storeframework, then I followed the structure already created at [storeframework.store](https://github.com/vtex-sites/storeframework.store/) to render things using CMS data.

### Regarding code architecture

**Since we want to easily remove CMS-related code from the repo**, I've added CMS components and pages in the same structure already used for non-cms stuff. i.e:
- index page without cms: `pages/index.tsx`
- index page with cms: `pages/cms/index.tsx`
- global alert component without cms: `components/common/Alert`
- global alert component with cms: `components/cms/common/Alert`

This way, it'd be easy to create a script that removes all cms-related code and overwrites it properly.

## How to test it?

Open the preview link. The global alert should be working with the message defined at the CMS. Which says: `Alert from master`. The link should send to `/office`.

You should also want to visit `/cms` and see if it's everything fine.


## References

https://www.faststore.dev/tutorials/cms-overview
https://github.com/vtex-sites/storeframework.store

## Checklist

<em>You may erase this after checking them all ;)</em>

**Changelog**

**PR Description**

**Documentation**
